code/modules/mod/mod_activation.dm

```diff
@@ -187,29 +187,28 @@
 
 /// Finishes the suit's activation, starts processing
 /obj/item/mod/control/proc/finish_activation(on)
-	icon_state = "[skin]-control[on ? "-sealed" : ""]"
-	slowdown = on ? slowdown_active : slowdown_inactive
-	if(on)
+	active = on
+	icon_state = "[skin]-control[active ? "-sealed" : ""]"
```

Comment:
move to update_icon_state ?

Reply:
update icon state doesnt run automatically and its not set up for modsuits, it could be done in a future pr probably